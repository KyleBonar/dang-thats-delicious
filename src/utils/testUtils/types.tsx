export interface MockLink {
  text: string;
  to: string;
  target: "_blank" | "_self";
}
