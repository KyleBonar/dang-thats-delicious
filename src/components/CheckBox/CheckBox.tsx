import * as React from "react";
import styled from "styled-components";
import Label from "../Label/Label";

interface CheckBoxProps {
  checked: boolean;
  id: string;
  value: string;
  label: string;
  onClick: (event: React.MouseEvent<HTMLInputElement>) => void;
  className?: string;
}

const CheckBox: React.SFC<CheckBoxProps> = props => {
  return (
    <div className={props.className}>
      <Label htmlFor={props.id}>
        <input
          defaultChecked={props.checked}
          type="checkbox"
          value={props.value}
          id={props.id}
          onClick={props.onClick}
        />
        {props.label}
      </Label>
    </div>
  );
};

const StyledCheckbox = styled(CheckBox)`
  display: inline-block;
  padding: 5px 1rem;
  box-sizing: border-box;

  input {
    margin-right: 5px;
  }

  label {
    padding: 10px 10px 10px 10px;
    background-color: ${props =>
      props.checked
        ? props.theme.checkBoxActiveBackgroundColor
        : props.theme.checkBoxInActiveBackgroundColor};
    color: ${props =>
      props.checked
        ? props.theme.checkBoxActiveTextColor
        : props.theme.checkBoxInActiveTextColor};
  }
`;

export default StyledCheckbox;
