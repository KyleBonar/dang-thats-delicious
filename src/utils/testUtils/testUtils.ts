import * as React from "react";
import casual from "casual";
import { ReactWrapper } from "enzyme";
import configureStore from "redux-mock-store";

import { MockCard } from "./types";
import { AppState } from "../../redux/configureStore";
import { ISauce, ISaucesState } from "../../redux/sauces/types";
import { FlashMessageProps } from "../../components/FlashMessage/FlashMessage";
import Flatn from "../Flatn/Flatn";
import { IUserState } from "../../redux/users/types";

export const ITERATION_SIZE = 32;
const REACT_REGEX = /react(\d+)?./i;

// seed it so we get consistent results
casual.seed(777);

const relatedSauce = () => ({
  name: casual.name,
  slug: casual.uuid
});

export const fakeSauce = (): ISauce => ({
  _id: casual.random_element([undefined, casual.integer(1, 500)]),
  _addedToStore: casual.random_element([undefined, casual.unix_time]), // Unix time (in seconds) added to redux store
  _full: casual.random_element([undefined, casual.boolean]), // Whether we have full sauce or partial
  _related: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(relatedSauce)
  ]), // List of related sauces
  isAdminApproved: casual.random_element([undefined, casual.boolean]),
  name: casual.name,
  ingredients: casual.string,
  author: casual.name,
  created: casual.unix_time, // Unix time (in seconds)
  types: casual.array_of_words(),
  maker: casual.name,
  description: casual.string,
  photo: casual.random_element([undefined, casual.url]),
  shu: casual.random_element([
    undefined,
    casual.integer(1, 500),
    casual.integer(1, 500).toString()
  ]),
  reviews: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(x => casual.word)
  ]),
  city: casual.random_element([undefined, casual.city]),
  state: casual.random_element([undefined, casual.state]),
  country: casual.random_element([undefined, casual.country]),
  slug: casual.random_element([undefined, casual.uuid])
});

export const fakeSaucesState = (): ISaucesState => {
  const sauces: ISauce[] = new Array(casual.integer(1, 15))
    .fill(null)
    .map(fakeSauce);
  const { allSlugs, bySlug } = Flatn.saucesArr({ sauces });

  return {
    allSlugs,
    bySlug,
    total: allSlugs.length,
    query: undefined,
    saucesWithNewestReviews: undefined,
    newest: undefined,
    featured: undefined,
    types: casual.array_of_words(),
    orders: ["Newest", "Name", "Times Reviewed", "Avg Rating"]
  };
};

const fakeUsersState = (): IUserState => ({
  self: casual.random_element([
    undefined,
    {
      token: casual.random_element([undefined, casual.uuid]),
      displayName: casual.random_element([undefined, casual.name]),
      avatarURL: casual.random_element([undefined, casual.url]),
      isAdmin: casual.random_element([undefined, casual.boolean])
    }
  ]),
  byDisplayName: undefined,
  allDisplayNames: undefined
});

export const fakeStore = () => {
  const storeConfig = configureStore([]);

  const store: AppState = {
    sauces: fakeSaucesState(),
    users: fakeUsersState(),
    reviews: {
      byReviewID: undefined,
      allReviewIDs: undefined
    }
  };

  return storeConfig(store);
};

export const fakeJSXElement = (): JSX.Element => {
  return casual.random_element([
    React.createElement("div", { key: casual.uuid }, casual.string),
    React.createElement("span", { key: casual.uuid }, casual.string)
  ]);
};

const fakeCard = (): MockCard => ({
  title: casual.string,
  description: casual.string,
  to: casual.url,
  showLink: casual.boolean,
  imageLink: casual.url,
  className: casual.string,
  anchorText: casual.string
});

const wait = (amount = 0) =>
  new Promise(resolve => setTimeout(resolve, amount));

function simulateInputChange(
  wrapper: ReactWrapper,
  name: string,
  value: string
) {
  wrapper
    // .find(`input[name='${name}']`)
    .simulate("change", { target: { name, value } });
}

const isClassComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    component.prototype &&
    !!component.prototype.isReactComponent
  );
};

// Ensure compatability with transformed code
const isFunctionComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    String(component).includes("return") &&
    !!String(component).match(REACT_REGEX) &&
    String(component).includes(".createElement")
  );
};

const isElement = (typeElement): boolean => {
  return React.isValidElement(typeElement);
};

const isComponent = (component): boolean => {
  return isClassComponent(component) || isFunctionComponent(component);
};

const isDOMTypeElement = (typeElement): boolean => {
  return isElement(typeElement) && typeof typeElement.type === "string";
};

const generateValidPassword = (MIN_PASSWORD_LENGTH: number = 8) => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(
    MIN_PASSWORD_LENGTH,
    3 * MIN_PASSWORD_LENGTH
  );

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

const generateInValidPassword = (MIN_PASSWORD_LENGTH: number = 8) => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(0, MIN_PASSWORD_LENGTH);

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

export {
  casual,
  simulateInputChange,
  isDOMTypeElement,
  isComponent,
  wait,
  generateValidPassword
};
