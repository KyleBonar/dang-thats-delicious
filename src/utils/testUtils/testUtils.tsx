import * as React from "react";
import * as enzyme from "enzyme";
import casual from "casual";
import { ReactWrapper } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import { AppState } from "../../redux/configureStore";
import { ISauce, ISaucesState } from "../../redux/sauces/types";
import Flatn from "../Flatn/Flatn";
import { IUserState } from "../../redux/users/types";
import { IErrReturn } from "../Err/Err";
import { act } from "react-dom/test-utils";
import { Provider } from "react-redux";

export const ITERATION_SIZE = 32;
const REACT_REGEX = /react(\d+)?./i;

// seed it so we get consistent results
casual.seed(777);

const relatedSauce = () => ({
  name: casual.name,
  slug: casual.uuid
});

export const fakeSauce = (): ISauce => ({
  _id: casual.random_element([undefined, casual.integer(1, 500)]),
  _addedToStore: casual.random_element([undefined, casual.unix_time]), // Unix time (in seconds) added to redux store
  _full: casual.random_element([undefined, casual.boolean]), // Whether we have full sauce or partial
  _related: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(relatedSauce)
  ]), // List of related sauces
  isAdminApproved: casual.random_element([undefined, casual.boolean]),
  name: casual.name,
  ingredients: casual.random_element([undefined, casual.string]),
  author: casual.name,
  created: casual.unix_time, // Unix time (in seconds)
  types: casual.random_element([undefined, casual.array_of_words()]),
  maker: casual.name,
  description: casual.string,
  photo: casual.random_element([undefined, casual.url]),
  shu: casual.random_element([
    undefined,
    casual.integer(1, 500),
    casual.integer(1, 500).toString()
  ]),
  reviews: casual.random_element([
    undefined,
    new Array(ITERATION_SIZE).fill(null).map(x => casual.word)
  ]),
  city: casual.random_element([undefined, casual.city]),
  state: casual.random_element([undefined, casual.state]),
  country: casual.random_element([undefined, casual.country]),
  slug: casual.random_element([undefined, casual.uuid])
});

export const generateFakeSauces = (): ISauce[] =>
  new Array(casual.integer(1, 15)).fill(null).map(fakeSauce);

export const fakeSaucesState = (): ISaucesState => {
  const sauces: ISauce[] = generateFakeSauces();
  const { allSlugs, bySlug } = Flatn.saucesArr({ sauces });

  const featured: undefined | string[] = casual.random_element([
    undefined,
    allSlugs.length > 0
      ? allSlugs.filter(slug => {
          if (casual.random < 0.4) return slug;
          return;
        })
      : undefined
  ]);

  return {
    allSlugs,
    bySlug,
    total: allSlugs.length,
    query: undefined,
    saucesWithNewestReviews: undefined,
    newest: undefined,
    featured,
    types: casual.array_of_words(),
    orders: ["Newest", "Name", "Times Reviewed", "Avg Rating"]
  };
};

const fakeUsersState = (): IUserState => ({
  self: casual.random_element([
    undefined,
    {
      token: casual.random_element([undefined, casual.uuid]),
      displayName: casual.random_element([undefined, casual.name]),
      avatarURL: casual.random_element([undefined, casual.url]),
      isAdmin: casual.random_element([undefined, casual.boolean])
    }
  ]),
  byDisplayName: undefined,
  allDisplayNames: undefined
});

export const fakeStore = () => {
  const middlewares = [thunk];
  const storeConfig = configureStore(middlewares);

  const store: AppState = {
    sauces: fakeSaucesState(),
    users: fakeUsersState(),
    reviews: {
      byReviewID: undefined,
      allReviewIDs: undefined
    }
  };

  return storeConfig(store);
};

export const fakeJSXElement = (): JSX.Element => {
  return casual.random_element([
    React.createElement("div", { key: casual.uuid }, casual.string),
    React.createElement("span", { key: casual.uuid }, casual.string)
  ]);
};

const wait = (amount = 0) =>
  new Promise(resolve => setTimeout(resolve, amount));

function simulateInputChange(
  wrapper: ReactWrapper,
  name: string,
  value: string
) {
  wrapper
    // .find(`input[name='${name}']`)
    .simulate("change", { target: { name, value } });
}

const isClassComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    component.prototype &&
    !!component.prototype.isReactComponent
  );
};

// Ensure compatability with transformed code
const isFunctionComponent = (component): boolean => {
  return (
    typeof component === "function" &&
    String(component).includes("return") &&
    !!String(component).match(REACT_REGEX) &&
    String(component).includes(".createElement")
  );
};

const isElement = (typeElement): boolean => {
  return React.isValidElement(typeElement);
};

const isComponent = (component): boolean => {
  return isClassComponent(component) || isFunctionComponent(component);
};

const isDOMTypeElement = (typeElement): boolean => {
  return isElement(typeElement) && typeof typeElement.type === "string";
};

const generateValidPassword = (MIN_PASSWORD_LENGTH: number = 8): string => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(
    MIN_PASSWORD_LENGTH,
    3 * MIN_PASSWORD_LENGTH
  );

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

const generateInValidPassword = (MIN_PASSWORD_LENGTH: number = 8): string => {
  // generate length between minimum length and 3x the minimum length
  const _randomLength = casual.integer(1, MIN_PASSWORD_LENGTH - 1);

  // turn generated length into random letters
  return new Array(_randomLength).fill(null).map(casual._letter).join("");
};

// Function designed to resemble how a network error would look like.
// This will help the catch(err) only have to handle a single format
const generateErr = (): IErrReturn => {
  return {
    response: {
      data: {
        status: 400,
        msg: casual.string,
        isGood: false
      }
    }
  };
};

// lets us mount up a simple custom hook
const mountReactHook = hook => {
  const Component = ({ children }) => children(hook());
  const componentHook = {};
  let componentMount;

  act(() => {
    componentMount = enzyme.shallow(
      <Component>
        {hookValues => {
          Object.assign(componentHook, hookValues);
          return null;
        }}
      </Component>
    );
  });
  return { componentMount, componentHook };
};

export interface ImountReactHookWithReduxStore {
  componentMount: any;
  componentHook: any;
}
const mountReactHookWithReduxStore = (
  hook,
  store
): ImountReactHookWithReduxStore => {
  const Component = ({ children }) => children(hook());
  const componentHook = {};
  let componentMount;

  act(() => {
    componentMount = enzyme.mount(
      <Provider store={store}>
        <Component>
          {hookValues => {
            Object.assign(componentHook, hookValues);
            return null;
          }}
        </Component>
      </Provider>
    );
  });
  return { componentMount, componentHook };
};

export {
  casual,
  simulateInputChange,
  isDOMTypeElement,
  isComponent,
  wait,
  generateValidPassword,
  generateInValidPassword,
  generateErr,
  mountReactHook,
  mountReactHookWithReduxStore
};
