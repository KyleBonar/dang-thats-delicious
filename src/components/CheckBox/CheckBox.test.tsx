import * as React from "react";
import * as enzyme from "enzyme";

import { casual } from "../../utils/testUtils/testUtils";
import CheckBox, { CheckBoxProps } from "./CheckBox";

const fakeCheckBox = (): CheckBoxProps => ({
  checked: casual.boolean,
  id: casual.uuid,
  value: casual.string,
  label: casual.string,
  onClick: jest.fn(),
  className: casual.random_element([undefined, casual.string])
});

describe("<CheckBox />", () => {
  it("renders", () => {
    const wrapper = enzyme.shallow(<CheckBox {...fakeCheckBox()} />);

    expect(wrapper).toBeTruthy();
  });

  it("matches snapshot", () => {
    const wrapper = enzyme.shallow(<CheckBox {...fakeCheckBox()} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("adds className to parent", () => {
    const mockCheckBox = fakeCheckBox();
    const wrapper = enzyme.shallow(<CheckBox {...mockCheckBox} />);

    expect(wrapper.first().prop("className")).toContain(mockCheckBox.className);
  });

  it("renders an input with type checkbox", () => {
    const mockCheckBox = fakeCheckBox();
    const wrapper = enzyme.shallow(<CheckBox {...mockCheckBox} />);

    expect(wrapper.find("input").props().type).toEqual("checkbox");
  });

  it("renders an input with expected id attribute", () => {
    const mockCheckBox = fakeCheckBox();
    const wrapper = enzyme.shallow(<CheckBox {...mockCheckBox} />);

    expect(wrapper.find("input").prop("id")).toEqual(mockCheckBox.id);
  });

  it("renders an input with expected checked attribute", () => {
    const mockCheckBox = fakeCheckBox();
    const wrapper = enzyme.shallow(<CheckBox {...mockCheckBox} />);

    expect(wrapper.find("input").prop("checked")).toEqual(mockCheckBox.checked);
  });

  it("renders a label element", () => {
    const mockCheckBox = fakeCheckBox();
    const wrapper = enzyme.shallow(<CheckBox {...mockCheckBox} />);

    expect(wrapper.find("label")).toBeTruthy();
  });

  it("renders a label with expected htmlFor attribute", () => {
    const mockCheckBox = fakeCheckBox();
    const wrapper = enzyme.shallow(<CheckBox {...mockCheckBox} />);

    expect(wrapper.find("Label").prop("htmlFor")).toEqual(mockCheckBox.id);
  });
});
