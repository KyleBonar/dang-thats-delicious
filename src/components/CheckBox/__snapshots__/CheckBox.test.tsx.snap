// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> matches snapshot 1`] = `
<div>
  <Label
    htmlFor="b430a73d-8e9b-4cce-96a5-0618965ffea6"
  >
    <input
      checked={true}
      id="b430a73d-8e9b-4cce-96a5-0618965ffea6"
      onClick={[MockFunction]}
      type="checkbox"
      value="soluta non omnis consequatur enim quia autem"
    />
    et modi tenetur amet modi reprehenderit omnis
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 2`] = `
<div
  className="sed sapiente soluta corporis commodi eaque nisi"
>
  <Label
    htmlFor="82fd03e8-18d9-4d81-898d-fec8184c94bf"
  >
    <input
      checked={true}
      id="82fd03e8-18d9-4d81-898d-fec8184c94bf"
      onClick={[MockFunction]}
      type="checkbox"
      value="laudantium molestias assumenda quasi adipisci mollitia iste"
    />
    itaque omnis debitis rem similique temporibus aliquam
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 3`] = `
<div
  className="corporis qui aut nobis quo consequatur harum"
>
  <Label
    htmlFor="50f67ee1-6558-4f56-9601-06bcd266da80"
  >
    <input
      checked={true}
      id="50f67ee1-6558-4f56-9601-06bcd266da80"
      onClick={[MockFunction]}
      type="checkbox"
      value="ex expedita est doloremque explicabo eaque necessitatibus"
    />
    sit suscipit omnis necessitatibus maiores ullam quae
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 4`] = `
<div
  className="nihil iste est unde rem similique et"
>
  <Label
    htmlFor="906618d0-0ced-482d-9e59-fdfec406f4f7"
  >
    <input
      checked={true}
      id="906618d0-0ced-482d-9e59-fdfec406f4f7"
      onClick={[MockFunction]}
      type="checkbox"
      value="sed consectetur soluta inventore laudantium dolor sint"
    />
    officiis qui dignissimos dolore qui voluptatem cumque
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 5`] = `
<div>
  <Label
    htmlFor="6b1588f4-b56b-4c68-896e-9d13d38d97d6"
  >
    <input
      checked={true}
      id="6b1588f4-b56b-4c68-896e-9d13d38d97d6"
      onClick={[MockFunction]}
      type="checkbox"
      value="sit saepe assumenda deserunt consequatur itaque et"
    />
    in fugit in rerum libero odit aut
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 6`] = `
<div
  className="eaque atque quos rerum nobis nam dolor"
>
  <Label
    htmlFor="6d81cafa-0cf4-4cc8-bf5a-77b9b1bb9cf9"
  >
    <input
      checked={true}
      id="6d81cafa-0cf4-4cc8-bf5a-77b9b1bb9cf9"
      onClick={[MockFunction]}
      type="checkbox"
      value="quis expedita et optio consequatur velit voluptate"
    />
    necessitatibus excepturi et qui sunt nesciunt provident
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 7`] = `
<div
  className="in eum ut dignissimos aspernatur voluptate voluptatem"
>
  <Label
    htmlFor="827928a0-2168-460c-86ee-001ad5580c98"
  >
    <input
      checked={false}
      id="827928a0-2168-460c-86ee-001ad5580c98"
      onClick={[MockFunction]}
      type="checkbox"
      value="quis quis nobis repudiandae aliquam molestiae occaecati"
    />
    a culpa facilis eligendi voluptatibus similique blanditiis
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 8`] = `
<div>
  <Label
    htmlFor="32521eb1-8c9d-4e4c-95c8-3ec38a2d7cbe"
  >
    <input
      checked={true}
      id="32521eb1-8c9d-4e4c-95c8-3ec38a2d7cbe"
      onClick={[MockFunction]}
      type="checkbox"
      value="aliquid ex architecto ut corrupti voluptatem quia"
    />
    dolores dolor voluptates sequi perferendis nostrum aliquid
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 9`] = `
<div>
  <Label
    htmlFor="5a3dfc83-0a85-4a82-8132-bf032630d26c"
  >
    <input
      checked={false}
      id="5a3dfc83-0a85-4a82-8132-bf032630d26c"
      onClick={[MockFunction]}
      type="checkbox"
      value="ipsam sed nobis excepturi saepe autem quis"
    />
    molestiae et dicta odit vel quisquam qui
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 10`] = `
<div>
  <Label
    htmlFor="f1b61de7-6e61-4957-ae52-f67e19477f3c"
  >
    <input
      checked={false}
      id="f1b61de7-6e61-4957-ae52-f67e19477f3c"
      onClick={[MockFunction]}
      type="checkbox"
      value="quasi quisquam ut libero explicabo at quae"
    />
    autem eligendi nihil quis eveniet sit voluptas
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 11`] = `
<div
  className="qui illo excepturi esse porro impedit omnis"
>
  <Label
    htmlFor="61511036-be53-4564-9aff-5d5dbe062ec7"
  >
    <input
      checked={true}
      id="61511036-be53-4564-9aff-5d5dbe062ec7"
      onClick={[MockFunction]}
      type="checkbox"
      value="esse qui est aut laborum nesciunt dolor"
    />
    vero voluptate esse aliquam molestiae quas omnis
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 12`] = `
<div
  className="sunt modi aut dolorem in non dolorem"
>
  <Label
    htmlFor="2b127fd3-b9e0-471a-b8f0-4fde2b658877"
  >
    <input
      checked={true}
      id="2b127fd3-b9e0-471a-b8f0-4fde2b658877"
      onClick={[MockFunction]}
      type="checkbox"
      value="esse dolores vitae delectus molestias ut distinctio"
    />
    molestias placeat exercitationem unde minus facere iste
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 13`] = `
<div>
  <Label
    htmlFor="a8072542-aacb-47ec-8d2f-d86fb60dc7d1"
  >
    <input
      checked={false}
      id="a8072542-aacb-47ec-8d2f-d86fb60dc7d1"
      onClick={[MockFunction]}
      type="checkbox"
      value="placeat ipsum assumenda officia aut id possimus"
    />
    quas dolor error eum incidunt provident deleniti
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 14`] = `
<div
  className="delectus aliquid et dicta ipsam qui provident"
>
  <Label
    htmlFor="0379b535-bacc-4578-afe3-c6659bca9a8c"
  >
    <input
      checked={false}
      id="0379b535-bacc-4578-afe3-c6659bca9a8c"
      onClick={[MockFunction]}
      type="checkbox"
      value="ratione qui voluptate sit excepturi sunt consectetur"
    />
    perspiciatis nisi repellendus quia voluptate optio suscipit
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 15`] = `
<div>
  <Label
    htmlFor="690ec9de-c290-435f-8a7a-f53c8fe7db13"
  >
    <input
      checked={false}
      id="690ec9de-c290-435f-8a7a-f53c8fe7db13"
      onClick={[MockFunction]}
      type="checkbox"
      value="facere recusandae ea necessitatibus sed ea ab"
    />
    doloremque est nemo iste omnis eaque nobis
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 16`] = `
<div
  className="eius est nihil sit enim itaque nostrum"
>
  <Label
    htmlFor="2af87018-a580-4816-b12e-1b33398f2d7f"
  >
    <input
      checked={false}
      id="2af87018-a580-4816-b12e-1b33398f2d7f"
      onClick={[MockFunction]}
      type="checkbox"
      value="distinctio libero mollitia facilis est laboriosam tempora"
    />
    aut cupiditate aut non ipsam fuga rem
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 17`] = `
<div>
  <Label
    htmlFor="945ab761-1b95-4891-b984-de2c04c155c1"
  >
    <input
      checked={false}
      id="945ab761-1b95-4891-b984-de2c04c155c1"
      onClick={[MockFunction]}
      type="checkbox"
      value="numquam ut et quia cupiditate minus cum"
    />
    est itaque qui dolorem voluptatem laboriosam iusto
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 18`] = `
<div>
  <Label
    htmlFor="15c773cf-e13b-427d-8f25-6705cf76fdd1"
  >
    <input
      checked={false}
      id="15c773cf-e13b-427d-8f25-6705cf76fdd1"
      onClick={[MockFunction]}
      type="checkbox"
      value="ut voluptate delectus officia maiores cumque consequuntur"
    />
    rerum aspernatur nulla aut omnis blanditiis accusamus
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 19`] = `
<div>
  <Label
    htmlFor="1c6179a7-21e8-4fab-914d-ec89bbb55dbd"
  >
    <input
      checked={false}
      id="1c6179a7-21e8-4fab-914d-ec89bbb55dbd"
      onClick={[MockFunction]}
      type="checkbox"
      value="doloribus aut maiores laborum soluta rerum atque"
    />
    velit voluptatem itaque quia accusantium autem consequuntur
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 20`] = `
<div>
  <Label
    htmlFor="f5069766-53f0-42f4-a572-bc0fa809cdeb"
  >
    <input
      checked={false}
      id="f5069766-53f0-42f4-a572-bc0fa809cdeb"
      onClick={[MockFunction]}
      type="checkbox"
      value="voluptatum qui exercitationem eum laudantium autem et"
    />
    recusandae voluptatem possimus voluptas eaque expedita iste
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 21`] = `
<div>
  <Label
    htmlFor="4f64a9a1-bcd1-4673-b49c-ee0bb155209d"
  >
    <input
      checked={false}
      id="4f64a9a1-bcd1-4673-b49c-ee0bb155209d"
      onClick={[MockFunction]}
      type="checkbox"
      value="facere fugiat rerum officiis assumenda nihil ad"
    />
    ut maxime aliquid sint voluptas saepe dignissimos
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 22`] = `
<div>
  <Label
    htmlFor="69f01164-3ca8-4cc8-912e-89def5301a86"
  >
    <input
      checked={true}
      id="69f01164-3ca8-4cc8-912e-89def5301a86"
      onClick={[MockFunction]}
      type="checkbox"
      value="ullam voluptatibus minima consequatur eligendi excepturi fugit"
    />
    beatae ut ea rerum nobis sit nihil
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 23`] = `
<div
  className="ut qui odit nam doloribus et facere"
>
  <Label
    htmlFor="4ed584a2-2d30-4828-8bf5-ba0501d49894"
  >
    <input
      checked={true}
      id="4ed584a2-2d30-4828-8bf5-ba0501d49894"
      onClick={[MockFunction]}
      type="checkbox"
      value="asperiores et consectetur animi non reiciendis iure"
    />
    laudantium est dolorem ratione sed enim provident
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 24`] = `
<div
  className="doloremque laudantium dolorum magni molestias dolorem nesciunt"
>
  <Label
    htmlFor="326b95e5-e2be-4f25-b9a5-84d7daa3874f"
  >
    <input
      checked={false}
      id="326b95e5-e2be-4f25-b9a5-84d7daa3874f"
      onClick={[MockFunction]}
      type="checkbox"
      value="dolor et placeat rerum sequi ex adipisci"
    />
    fugiat officiis enim voluptatem adipisci sequi magnam
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 25`] = `
<div>
  <Label
    htmlFor="b395f68f-198d-4b09-a257-ed4482fd54e5"
  >
    <input
      checked={true}
      id="b395f68f-198d-4b09-a257-ed4482fd54e5"
      onClick={[MockFunction]}
      type="checkbox"
      value="suscipit sit et culpa cum architecto nulla"
    />
    ut molestias nisi corporis nulla officia rerum
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 26`] = `
<div
  className="sed est et sapiente ipsa aut nostrum"
>
  <Label
    htmlFor="65b0eef9-89a9-4dc2-8726-bdad1b901852"
  >
    <input
      checked={true}
      id="65b0eef9-89a9-4dc2-8726-bdad1b901852"
      onClick={[MockFunction]}
      type="checkbox"
      value="ullam ullam doloremque quidem vel voluptatem ab"
    />
    incidunt velit voluptatem pariatur blanditiis voluptatem laboriosam
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 27`] = `
<div>
  <Label
    htmlFor="3f799156-3acb-4f48-9507-990989db8977"
  >
    <input
      checked={false}
      id="3f799156-3acb-4f48-9507-990989db8977"
      onClick={[MockFunction]}
      type="checkbox"
      value="nihil voluptatem est impedit assumenda qui cum"
    />
    quos eos iste aut inventore optio tempora
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 28`] = `
<div>
  <Label
    htmlFor="bae84085-e845-4a9b-b610-4457b228f993"
  >
    <input
      checked={true}
      id="bae84085-e845-4a9b-b610-4457b228f993"
      onClick={[MockFunction]}
      type="checkbox"
      value="saepe quam quia veritatis qui optio autem"
    />
    et dolorum explicabo provident molestiae qui est
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 29`] = `
<div
  className="labore et ut unde consequatur sequi autem"
>
  <Label
    htmlFor="d97e0345-df64-4a08-b3e9-c2a1d4a43630"
  >
    <input
      checked={false}
      id="d97e0345-df64-4a08-b3e9-c2a1d4a43630"
      onClick={[MockFunction]}
      type="checkbox"
      value="libero ipsa voluptates quaerat recusandae accusamus et"
    />
    unde quia incidunt ex et molestiae facere
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 30`] = `
<div>
  <Label
    htmlFor="75985099-e671-4c57-ad21-616b1e71168e"
  >
    <input
      checked={true}
      id="75985099-e671-4c57-ad21-616b1e71168e"
      onClick={[MockFunction]}
      type="checkbox"
      value="perspiciatis nulla maiores pariatur amet harum maxime"
    />
    a repellendus perspiciatis vero dolores incidunt est
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 31`] = `
<div>
  <Label
    htmlFor="90aa5c30-2851-4973-bc77-d691f5902ed9"
  >
    <input
      checked={false}
      id="90aa5c30-2851-4973-bc77-d691f5902ed9"
      onClick={[MockFunction]}
      type="checkbox"
      value="eligendi dicta sit cum voluptatem dolor delectus"
    />
    unde reprehenderit deleniti nihil ex maiores nihil
  </Label>
</div>
`;

exports[`<CheckBox /> matches snapshot 32`] = `
<div
  className="facilis similique optio hic perspiciatis dolor quod"
>
  <Label
    htmlFor="1f75c79b-ef69-42be-9fa7-b35efba5b409"
  >
    <input
      checked={true}
      id="1f75c79b-ef69-42be-9fa7-b35efba5b409"
      onClick={[MockFunction]}
      type="checkbox"
      value="aliquam consequuntur omnis velit quibusdam aut in"
    />
    blanditiis quia aliquam eum exercitationem excepturi eligendi
  </Label>
</div>
`;
